managed/ui/src/components/universes/UniverseForm/AZPlacementInfo.js

```diff
@@ -53,48 +53,53 @@ export default class AZPlacementInfo extends Component {
     placementInfo: PropTypes.object.isRequired
   };
   render() {
-    const { placementInfo, placementCloud } = this.props;
-    if (!isNonEmptyObject(placementInfo) || !isNonEmptyObject(placementCloud)) {
+    const { placementInfo, placementCloud, providerCode } = this.props;
+    let currentStatusType;
+
+    // If placementInfo and placementCloud is empty and provier is not onprem then return
+    if ((!isNonEmptyObject(placementInfo) || !isNonEmptyObject(placementCloud)) && providerCode !== "onprem") {
       return <span />;
     }
 
     const replicationFactor = placementInfo.replicationFactor;
-    const regionList = placementCloud.regionList;
+    const regionList = placementCloud?.regionList;
     let multiRegion = true;
     let multiAz = true;
 
     // This logic is to help determine whether any AZ or region in the current universe config
     // contains a majority of tablet replicas. This determines whether the current config will
     // result in a cluster that is resilient to AZ or Region level failures while still maintaining
     // quorum.
-    regionList.forEach((region) => {
-      const azList = region.azList;
-      let regionNumReplicas = 0;
-      azList.forEach((az) => {
-        regionNumReplicas += az.replicationFactor;
+    if (regionList) {
+      regionList.forEach((region) => {
+        const azList = region.azList;
+        let regionNumReplicas = 0;
+        azList.forEach((az) => {
+          regionNumReplicas += az.replicationFactor;
+          if (replicationFactor % 2 === 0) {
+            if (replicationFactor / 2 < az.replicationFactor) {
+              multiAz = false;
+            }
+          } else {
+            if ((replicationFactor - 1) / 2 < az.replicationFactor) {
+              multiAz = false;
+            }
+          }
+        });
```

Comment:
@daniel-yb pls take a look, I remember it's you added the logic above

Reply:
@daniel-yb Please add your inputs here.

Reply:
@yb-andrew / @sshev you'd know this better than I do, but aren't all numbers in javascript effectively a double? I guess you could use `Math.floor(replicationFactor / 2) < az.replicationFactor` or something like this to achieve the same behavior while simplifying the two cases of even/odd here. Also now that I think of it, since az.replicationFactor will always be a whole number, we don't even really need the if/else at all and can go with Andrew's recommendation. Otherwise this change looks fine to me; it shouldn't change the results of the calculations here, it should only skip them if the regionList is not null. You could also just do `const regionList = _.get(placementCloud, "regionList", []);` above instead of adding this new `if (regionList) {...}` case (which IMO would be cleaner since it doesn't indent a whole block of code under another if statement).

Reply:
@daniel-yb Numbers in Javascript can be int, floats, unsigned ints, unsigned floats, or big ints. You don't need the Math.floor if the `replicationFactor` is an int because dividing by another int will truncate the value (e.g. 5 / 2 = 2).
@gaurav061 You can use the suggestion of simplifying the even/odd logic cases but just add a code comment for clarity that `replicationFactor` is an int.

Reply:
@daniel-yb I'm incorrect here. Numbers in JS default to double like you said unless initializing with a different class (unsigned, bigint).